administrator/components/com_content/models/article.php

```diff
@@ -706,6 +705,53 @@ public function featured($pks, $value = 0)
 	}
 
 	/**
+	 * Method to store the token generated.
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @since 3.7
+	 */
+	private function shareTokenGenerate()
+	{
+		jimport('joomla.user.helper');
+		$token = JUserHelper::genRandomPassword(16);
+
+		return $token;
+	}
+
+	/**
+	 * Method to store the token generated.
+	 *
+	 * @param   string  $title  The title of the shared article.
+	 *
+	 * @return  boolean  True on success.
```

Comment:
Imo the method should return the generated token (as you are going to need that after the ajax call) and not the boolean.

/cc @roland-d you know more about the planed implementation. Can you comment?